src/Raven.Studio/typescript/models/database/index/additionalAssemblyModel.ts

```diff
@@ -0,0 +1,182 @@
+/// <reference path="../../../../typings/tsd.d.ts"/>
+import jsonUtil = require("common/jsonUtil");
+
+type assemblySource = "Server Runtime" | "Path" | "NuGet";
+
+class additionalAssemblyModel {
+    
+    assemblySource = ko.observable<assemblySource>();
+    
+    assemblyName = ko.observable<string>();
+    assemblyPath = ko.observable<string>();
+    
+    packageName = ko.observable<string>();
+    packageVersion = ko.observable<string>();
+    packageSourceUrl = ko.observable<string>();
+    useDefaultPackageSourceUrl = ko.observable<boolean>();
+    
+    usings = ko.observableArray<string>();
+    namespaceText = ko.observable<string>();
+
+    dirtyFlag: () => DirtyFlag;
+    validationGroup: KnockoutObservable<any>;
+    
+    constructor(dto: Raven.Client.Documents.Indexes.AdditionalAssembly) {
+        this.assemblyName(dto.AssemblyName);
+        this.assemblyPath(dto.AssemblyPath);
+        
+        this.packageName(dto.PackageName);
+        this.packageVersion(dto.PackageVersion);
+        
+        this.packageSourceUrl(dto.PackageSourceUrl);
+        this.useDefaultPackageSourceUrl(!dto.PackageSourceUrl)
+        
+        this.usings(dto.Usings);
+
+        this.assemblySource(this.computeAssemblySource());
+       
+        this.initObservables();
+        this.initValidation();
+    }
+
+    private initObservables() {
+        this.useDefaultPackageSourceUrl.subscribe((useDefault) => {
+            if (useDefault) {
+                this.packageSourceUrl(null);
+            }
+        });
+        
+        this.assemblySource.subscribe((newSource) => {
+            this.usings([]);
+            
+            switch (newSource) {
+                case "Server Runtime":
```

Comment:
I wonder if we want to clear those fields or just ignore them when saving?

Reply:
done - cleared relevant fields only upon save